### PR TITLE
.codespell*,.github,*: add codespell configuration, workflow, fix codespell errors.

### PR DESCRIPTION
**Pull request by klihub**
This PR 
- adds codespell configuration
- adds a codespell workflow
- fixes existing spelling errors detected by codespell

This PR will conflict with #332 if/when that gets merged. I'd prefer us to focus our limited review capacity on #332, Also, I expect reviewing this to mostly be a one-time skim-through. Therefore, I'll keep this a draft until #332 gets merged.

### CODE REVIEW

File: .codespellrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[codespell]
+skip = .git,*.pdf,*.svg,go.sum,go.mod
+ignore-words = .codespellignorewords
+exclude-file = .codespellignorelines
```
Comment (fmuyassarov) [inline .codespellrc:4]: missing a new line

**Review by fmuyassarov — COMMENTED** [review]
Do you think it is okay to merge all the commits that fixing typos ? That way we would end up with three commits one for the actual makefile target, one for GitHub Actions and the last one fixing the typos.

**Review by fmuyassarov — APPROVED** [review]
LGTM

**Comment by fmuyassarov** [comment]
ping @askervin

**Review by kad — APPROVED** [review]
LGTM

**Review by askervin — APPROVED** [review]
LGTM. Great to get these fixed now and in the future. :D

**Pull request merged**